setup.cfg

```diff
@@ -91,7 +91,7 @@ install_requires =
     # Cattrs upgrades were known to break lineage https://github.com/apache/airflow/issues/16172
     # TODO: Cattrs is now at 3.8 version so we should attempt to upgrade cattrs soon.
     cattrs~=1.1, !=1.7.*
-    colorlog>=4.0.2
+    colorlog>=4.0.2, <5.0
```

Comment:
We should describae why we limited it - which tests failed - ideally what were the proble and what we need to fix it. 

Or maybe we can fix it ? I doubt colorlog has some "really big" changes.

Reply:
It felt fixable when I first spotted it yesterday - the issues all seemed to be related to `CustomTTYColoredFormatter` but it didn't feel right to make changes to Airflow logging as part of this PR, as it's totally unrelated to Google Ads

It was actually mypy that threw up errors - the `TTYColoredFormatter` class we inherit from was merged into `ColoredFormatter`, and as a result, a few functions have changed names / signatures.

Can introduce a comment as part of this PR, and can look to upgrade it in a separate PR

Reply:
Separate PR is way better. Just comment in this one as we have a rule thall all upper-bounded limitations have to have comment why and what is needed to lift it.

> [](https://github.com/apache/airflow/blob/main/README.md#approach-for-dependencies-for-airflow-core)Whenever we upper-bound such a dependency, we should always comment why we are doing it - i.e. we should have a good reason why dependency is upper-bound. And we should also mention what is the condition to remove the binding.